src/write_sensu.c

```diff
@@ -39,6 +39,53 @@
 #include <stddef.h>
 
 #include <stdlib.h>
+#ifdef HAVE_KERNEL_SOLARIS
```

Comment:
Yeah, it should, good point.

> The problem is that the machine where I compile the package has vasprintf but I disallow the linker to bind against the libc contract that provides it.

Why do you do this? Is this machine running Solaris 10 update 11 and are you trying to emulate update 8?
Did configure determine that asprintf was available?